CoroutineTemplate/data/src/main/java/co/nimblehq/coroutine/data/service/response/UserResponse.kt

```diff
@@ -24,3 +25,36 @@ data class UserResponse(
         )
     }
 }
+
+fun List<UserResponse>.toUserEntities(): List<UserEntity> {
+    return this.map { it.toEntity() }
+}
+
+private fun UserResponse.toEntity(): UserEntity {
+    return UserEntity(
+        id = this.id,
+        name = this.name.orEmpty(),
+        username = this.username.orEmpty(),
+        email = this.email.orEmpty(),
+        address = this.address?.toEntity(),
+        phone = this.phone.orEmpty(),
+        website = this.website.orEmpty()
+    )
```

Comment:
```suggestion
    return with(this) {
        UserEntity(
            id = id,
            name = name.orEmpty(),
            username = username.orEmpty(),
            email = email.orEmpty(),
            address = address?.toEntity(),
            phone = phone.orEmpty(),
            website = website.orEmpty()
        )
    }
```
Can we use the scope function? What do you think?

Reply:
@lydiasama I think it is a redundant step, just make our code more complicated without having more benefit. 🤔

Reply:
Hmm. true 🤔  Actually we can get rid of `this` because it can directly call the properties inside of the parent class also. But for the extension function, I think to keep the `this` is better to ensure it will use the correct `this` always. 

I just feel it has multiple `this` and it's quite long on the right side. It's just an optional. I'm ok. 👌🏻

Reply:
@AVI5HEK Hmm 🤔  I agree at this point. `id = this.id` is clearer.